addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java

```diff
@@ -568,57 +568,69 @@ public synchronized void playURI(String uri) {
+            return socket.callMethod(method, params);
+        });
 
+        List<KodiPVRChannelGroup> channelGroups = new ArrayList<>();
         if (response instanceof JsonObject) {
             JsonObject result = response.getAsJsonObject();
-            if (result.has("channelgroups")) {
-                return result.get("channelgroups").getAsJsonArray();
+            for (JsonElement element : result.get("channelgroups").getAsJsonArray()) {
+                JsonObject object = (JsonObject) element;
+                KodiPVRChannelGroup channelGroup = new KodiPVRChannelGroup();
+                channelGroup.setId(object.get("channelgroupid").getAsInt());
+                channelGroup.setLabel(object.get("label").getAsString());
+                channelGroup.setChannelType(channelType);
+                channelGroups.add(channelGroup);
             }
         }
-        return null;
+        return channelGroups;
     }
 
     public int getChannelGroupID(final String channelType, final String channelGroupName) {
-        JsonArray channelGroups = getChannelGroups(channelType);
-        if (channelGroups instanceof JsonArray) {
-            for (JsonElement element : channelGroups) {
-                JsonObject channelGroup = (JsonObject) element;
-                String label = channelGroup.get("label").getAsString();
-                if (StringUtils.equalsIgnoreCase(label, channelGroupName)) {
-                    return channelGroup.get("channelgroupid").getAsInt();
-                }
+        List<KodiPVRChannelGroup> channelGroups = getChannelGroups(channelType);
+        for (KodiPVRChannelGroup channelGroup : channelGroups) {
+            if (StringUtils.equalsIgnoreCase(channelGroup.getLabel(), channelGroupName)) {
+                return channelGroup.getId();
             }
         }
         return 0;
     }
 
-    private synchronized JsonArray getChannels(final int channelGroupID) {
-        JsonObject params = new JsonObject();
-        params.addProperty("channelgroupid", channelGroupID);
-        JsonElement response = socket.callMethod("PVR.GetChannels", params);
+    public synchronized List<KodiPVRChannel> getChannels(final int channelGroupID) {
+        String method = "PVR.GetChannels";
+        String hash = method + "#channelgroupid=" + channelGroupID;
+        JsonElement response = REQUEST_CACHE.putIfAbsentAndGet(hash, () -> {
+            JsonObject params = new JsonObject();
+            params.addProperty("channelgroupid", channelGroupID);
+            return socket.callMethod(method, params);
+        });
 
+        List<KodiPVRChannel> channels = new ArrayList<>();
         if (response instanceof JsonObject) {
             JsonObject result = response.getAsJsonObject();
-            if (result.has("channels")) {
-                return result.get("channels").getAsJsonArray();
+            for (JsonElement element : result.get("channels").getAsJsonArray()) {
```

Comment:
Same comment as above concerning the assumption that the response always will contain the member `channels`.